hazelcast/src/hazelcast/client/compact.cpp

```diff
@@ -15,8 +15,1580 @@
+        BOOST_THROW_EXCEPTION(exception::hazelcast_serialization{ boost::str(
+          boost::format(
+            "Invalid field kind: '%1%' for %2%, expected : %3%, given : %4%") %
+          field_name % schema % fd->kind % kind) });
+    }
+}
+
+generic_record generic_record_builder::build()
+{
+    if (strategy_ == strategy::normal) {
+        pimpl::schema_writer& writer =
+          boost::get<pimpl::schema_writer>(writer_or_schema_);
+
+        already_built_ = true;
+        return generic_record{ std::move(writer).build(), std::move(objects_) };
+    } else {
+        pimpl::schema& schema = boost::get<pimpl::schema>(writer_or_schema_);
+
+        if (strategy_ == strategy::schema_bounded) {
+            const auto& fields = schema.fields();
+
+            for (const auto& p : fields) {
+                const std::string& field_name = p.first;
+
+                if (objects_.find(field_name) == end(objects_)) {
+                    BOOST_THROW_EXCEPTION(exception::hazelcast_serialization{
+                      "Found an unset field " + field_name +
+                      ". All the fields must be set before build" });
+                }
+            }
+        }
+
+        already_built_ = true;
+        return generic_record{ std::move(schema), std::move(objects_) };
+    }
+}
+
+generic_record_builder&
+generic_record_builder::set_boolean(std::string field_name, bool value)
+{
+    return write(move(field_name), value, field_kind::BOOLEAN);
+}
+
+generic_record_builder&
+generic_record_builder::set_int8(std::string field_name, int8_t value)
+{
+    return write(move(field_name), value, field_kind::INT8);
+}
+
+generic_record_builder&
+generic_record_builder::set_int16(std::string field_name, int16_t value)
+{
+    return write(move(field_name), value, field_kind::INT16);
+}
+
+generic_record_builder&
+generic_record_builder::set_int32(std::string field_name, int32_t value)
+{
+    return write(move(field_name), value, field_kind::INT32);
+}
+
+generic_record_builder&
+generic_record_builder::set_int64(std::string field_name, int64_t value)
+{
+    return write(move(field_name), value, field_kind::INT64);
+}
+
+generic_record_builder&
+generic_record_builder::set_float32(std::string field_name, float value)
+{
+    return write(move(field_name), value, field_kind::FLOAT32);
+}
+
+generic_record_builder&
+generic_record_builder::set_float64(std::string field_name, double value)
+{
+    return write(move(field_name), value, field_kind::FLOAT64);
+}
+
+generic_record_builder&
+generic_record_builder::set_nullable_boolean(std::string field_name,
+                                             boost::optional<bool> value)
+{
+    return write(move(field_name), value, field_kind::NULLABLE_BOOLEAN);
+}
+
+generic_record_builder&
+generic_record_builder::set_nullable_int8(std::string field_name,
+                                          boost::optional<int8_t> value)
+{
+    return write(move(field_name), value, field_kind::NULLABLE_INT8);
```

Comment:
"std::move" instead of "move" can be used.

Reply:
ADL handles it, I do it intentionally to draw an attraction of reader :D It is also less verbose.